src/autocomplete.js

```diff
@@ -709,6 +710,11 @@ class CompletionProvider {
 
     detach() {
         this.active = false;
+        this.completers && this.completers.forEach(function(completer) {
+            if (typeof completer.cancel === "function") {
+                completer.cancel();
```

Comment:
`completer.cancel` is simple to use for the Ace internals, but can be painful to Ace users. They will need to track the requests on the side of their custom completers - e.g. probably create custom AbortSignals inside their implementation of `getCompletions` and store them somewhere on the side just so they can cancel them later. Can we consider integration with AbortSignal on the `getCompletions` level? More logic on the Ace side, but should be much more ergonomic to the Ace users.

Reply:
How would abortSignal work here? I thought that cancel will have to simply make a new  [cancelRequest](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#cancelRequest) with previous requests id.